handlers.go

```diff
@@ -0,0 +1,363 @@
+	var task Task
+	switch {
+	case r.Method == http.MethodPost:
+		fmt.Print("INFO API task received POST message \"/api/task\"\n")
+		err := json.NewDecoder(r.Body).Decode(&task)
+		if err != nil {
+			fmt.Print("ERROR API unable to deserialize JSON  \"/api/task\"\n")
+			http.Error(w, `{"error": "JSON deserialization"}`,
+				http.StatusBadRequest)
+		}
+		fmt.Printf("DEBUG API POST message \"/api/task\" serialization %v\n",
+			task)
+		if task.Title == "" {
+			fmt.Print("ERROR API title is mandatory  \"/api/task\"\n")
+			http.Error(w, `{"error": "Title is required"}`,
+				http.StatusBadRequest)
+			return
+		}
+		if task.Date == "" {
+			fmt.Print("DEBUG API task date is missing  \"/api/task\"\n")
+			task.Date = time.Now().Format(dateFormat)
+		} else {
+			parsedDate, err := time.Parse(dateFormat, task.Date)
+			if err != nil {
+				http.Error(w, `{"error": "Invalid date format"}`,
+					http.StatusBadRequest)
+				return
+			}
+			//if parsedDate.Before(time.Now()) {
+			if parsedDate.Format(dateFormat) < time.Now().Format(dateFormat) {
+				fmt.Print("DEBUG API task date from \"/api/task\" is BEFORE NOW\n")
+				fmt.Printf("Parsed: %s, Now: %s\n", parsedDate.Format(dateFormat), time.Now().Format(dateFormat))
+				if task.Repeat == "" {
+					task.Date = time.Now().Format(dateFormat)
+				} else {
+					nextDate, err := NextDate(time.Now(), task.Date, task.Repeat)
+					if err != nil {
+						http.Error(w, fmt.Sprintf(`{"error": "Invalid repeat rule: %s"}`, err),
+							http.StatusBadRequest)
+						return
+					}
+					task.Date = nextDate
+				}
+			}
+		}
+		query := `INSERT INTO scheduler (date, title, comment, repeat) VALUES (?, ?, ?, ?)`
+		fmt.Printf("INFO SQL attempt to submit data to scheduler (%s, %s, %s, %s)\n",
+			task.Date, task.Title, task.Comment, task.Repeat)
+		result, err := db.Exec(query, task.Date, task.Title, task.Comment, task.Repeat)
+		if err != nil {
+			fmt.Printf("ERROR SQL unable to submit data to scheduler (%s, %s, %s, %s)\n",
+				task.Date, task.Title, task.Comment, task.Repeat)
+			http.Error(w, fmt.Sprintf(`{"error":"Unable to submit data to DB: %v"}`, err),
+				http.StatusInternalServerError)
+			return
+		}
+		id, err := result.LastInsertId()
+		if err != nil {
+			fmt.Print("ERROR SQL unable to get last scheduler task ID")
+			http.Error(w, fmt.Sprintf(`{"error":"Unable to get last scheduler task ID: %s"}`, err),
+				http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode(map[string]interface{}{"id": id})
+	case r.Method == http.MethodGet:
+		fmt.Print("INFO API task received GET message \"/api/task\"\n")
+		taskId := r.URL.Query().Get("id")
+		//fmt.Print(r.URL)
+		fmt.Printf("DEBUG API \"/api/task\" GET param \"id\": %s\n", taskId)
+		if len(taskId) == 0 {
+			fmt.Print("ERROR API incorrect task id parameter \"/api/task\"\n")
```

Comment:
Могу немного упростить тебе жизнь. Вот тебе 3 варинта. А вообще если очень интересно, то можешь посмотреть в сторону проф логеров, где путь можно как параметр передавать.
```suggestion
			fmt.Print("ERROR API incorrect task id parameter '/api/task'\n")
```.

```suggestion
			fmt.Print(`ERROR API incorrect task id parameter "/api/task"\n`)
```

```suggestion
			fmt.Println(`ERROR API incorrect task id parameter "/api/task"`)
```